### PR TITLE
vmpk: 0.5.1 -> 0.7.2

### PR DESCRIPTION
**Pull request by freezeboy**
Also migrate from qt4 to qt5 (#33248)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102521` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vmpk</li>
  </ul>
</details>

**Comment by jonringer** [comment]
I'm assuming I need some service or hardware to use this?

```
[nix-shell:/home/jon/.cache/nixpkgs-review/pr-102521]$ ./results/vmpk/bin/vmpk
Failure loading Qt5 translations for "en_US" from "/nix/store/dw26fwjr4lgwig2daikfagrpw4w2x8y5-qtbase-5.15.0/translations"
Failure loading VMPK translations for "en_US" from "/nix/store/g45ww56yrw0qjg2vrb764050hh6bdiyd-vmpk-0.7.2/bin/../share/vmpk/locale/"
Default MIDI IN driver "Network" not found
Default MIDI OUT driver "SonivoxEAS" not found
Unable to initialize all MIDI drivers. Terminating.
```

**Comment by freezeboy** [comment]
I guess also I had the same output but I don't have any midi hardware

**Comment by jonringer** [comment]
cc @orivej

File: pkgs/applications/audio/vmpk/default.nix [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
     url = "mirror://sourceforge/${pname}/${version}/${pname}-${version}.tar.bz2";
-    sha256 = "11fqnxgs9hr9255d93n7lazxzjwn8jpmn23nywdksh0pb1ffvfrc";
+    sha256 = "5oLrjQADg59Mxpb0CNLQAE574IOSYLDLJNaQ/9q2cMQ=";
   };
 
-  nativeBuildInputs = [ cmake pkgconfig ];
+  nativeBuildInputs = [ cmake pkg-config qttools ];
 
-  buildInputs = [ alsaLib libjack2 qt4 ];
+  buildInputs = [
```
Comment (orivej) [inline pkgs/applications/audio/vmpk/default.nix:18]: Please ignore CMake/pkg-config warnings and do not add static dependencies to buildInputs. ([Issue report](https://gitlab.kitware.com/cmake/cmake/-/issues/18158).) This seems OK:

    nativeBuildInputs = [ cmake docbook-xsl-nons pkg-config qttools ];
    buildInputs = [ drumstick qtx11extras ];

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -24106,7 +24106,7 @@ in
     onlyLibVLC = true;
   };
 
-  vmpk = callPackage ../applications/audio/vmpk { };
+  vmpk = qt5.callPackage ../applications/audio/vmpk { inherit (libsForQt5) drumstick; };
```
Comment (orivej) [inline pkgs/top-level/all-packages.nix:24109]: Please use libsForQt5.callPackage to provide drumstick (`vmpk = libsForQt5.callPackage ../applications/audio/vmpk { };`).

**Review by orivej — CHANGES_REQUESTED** [review]
vmpk does not need MIDI hardware, it only needs ALSA MIDI sequencer. (If `aplaymidi -l` does not give an error it should work. The old vmpk should work as well.)

vmpk 0.7.2 seems to [require](https://sourceforge.net/p/vmpk/bugs/53/) drumstick with libpulseaudio — otherwise drumstick does not ship libdrumstick-rt-eassynth.so — but even with this plugin installed drumstick seems unable to find it. This requires further investigation…

**Comment by orivej** [comment]
Thanks!

Drumstick plugins dir is supposed to be added to the search path [here](https://sourceforge.net/p/drumstick/code/471/tree/trunk/library/rt/backendmanager.cpp#l125). I have not checked why it does not work, but I've noticed a workaround [above](https://sourceforge.net/p/drumstick/code/471/tree/trunk/library/rt/backendmanager.cpp#l116) — executable-relative lookup. To exploit it, add `ln -s ${drumstick}/lib $out` to vmpk `postInstall` (and add libpulseaudio to drumstick `buildInputs`).

**Review by orivej — APPROVED** [review]
Thank you!

**Comment by orivej** [comment]
I've decided to patch drumstick to fix plugin lookup.

**Pull request merged**